### DISCUSSION TITLE
Make index_page configurable

### DISCUSSION THREAD

**Discussion by noelmcloughlin** [mkdocs/mkdocs#3632] [Ideas]
I would like  `index_page: <readme-or-some-other-name>.md` feature. While https://github.com/mkdocs/mkdocs/issues/608 is concerned with readme.md I see different problem. Index is a semantic concept in various [ontologys](https://www.ebi.ac.uk/ols4/search/index?exactMatch=false&includeObsoleteEntities=false&isDefiningOntology=true) and also wikidata KG ([here](https://www.wikidata.org/wiki/Q873506), [here](https://www.wikidata.org/wiki/Q1738991), [here](https://www.wikidata.org/wiki/Q580427)).

The [LinkML](https://linkml.io/) (life sciences mainly) community model concepts in schemas for various domains. Today I was modeling `index` slot/propety and mkdocs generates two `index.md` and second overwrites first. Since the slot/property **index** is entrenched in various ontologies -  using another name  is not possible.

Please add  flexibility for `index.md` to support semantic use cases, where "index" is a _**thing**_ outside mkdocs project.

**Comment by lovelydinosaur** [comment]
I'm going to push this into a discussion, rather than an issue for now.

Perhaps you could expand on exactly what behavior you're asking for here. We use `index.md` because it maps to `index.html` which is what webservers expect.

**Comment by noelmcloughlin** [comment]
Project subject matter may deal with concepts like '[index](https://www.wikidata.org/wiki/Q873506)' (and possibly '[readme](https://www.wikidata.org/wiki/Q539662)'). For example, my project redefined "[collections ontology:**index**](https://semantic-web-journal.net/system/files/swj432.pdf) -  but as workaround I had to rename to "[_index](https://lmodel.github.io/collections/_index/)" just to move on.

"index.html" is clearly web convention, mentioned in [rfc7231](https://datatracker.ietf.org/doc/html/rfc7231), but "index" is [ontology convention](https://www.ebi.ac.uk/ols4/search?q=index&exactMatch=true&includeObsoleteEntities=true) and ontology (owl/rdf) is semantic web standard.  I know ontology is niche use case (mainly used by medical science/academia) but I hoped it could be considered valid use case - out of mkdocs user's control - and supported.

**Comment by klvs** [comment]
I'd like to add my support for this. Let me explain my use case and reasoning:

I work at a large company with very fragment documentation. Like several thousand developers large (this isn't a boast, it's the scale of the problem). We're trying to corral all of the docs into a _somewhat centralized location_. This has proven very difficult since people use a lot of different tools.

One popular way of writing docs, at least within my company, is to just write them in markdown on github. Since this is a pretty sane and reasonable way of doing it, I want to support it. It's very difficult to convince teams to change the way they write docs, so I kind of have to support their way of writing docs if it at least makes sense.

Here's what I'm doing with regard to github docs: for teams that want to write docs in GH, I find their markdown docs repo, configure a jenkins job or github action, build them, and then publish them to a central location. You can imagine docs.company.com.

I'm evaluating different tools to support the teams that don't already have a docs tool (like docusaurus, gitbook, etc) and instead prefer to write their docs directly in github. This is how I stumbled on MkDocs and **oh my god thank you guys this is exactly what I need** for these teams.  

**One small problem** 
Usually these teams organize their docs in one of two ways: For teams that work on a single app, code base or project, they tend to put their docs inside their repo along side their code:

```
/
/.github
/src
/docs
/README.md
```
However, for teams that manage many different apps or code bases, they tend to instead desire a completely separate repo for all of their docs. Or they have docs along side their apps and a completely separate docs-only repo for their general team docs. Either way, you end up with a case where there are a lot of docs-only repositories:
```
/.github
/explanations
/how-to-guides
/tutorials
/references
/README.md
```

Due to github's repository defaults it's very common to use README.md very similarly to how MkDocs expects index.md. I understand your reasoning, but this convention is extremely common not just within my company, but github in general. In the world of github and git-based docs, README.md is the defacto index.html.

I can try to convince these teams that they need index.md, but ultimately it's going to be a LOT easier if I can just set a config option `index_page: README.md`. Especially when I may potentially be bootstraping several hundred repositories with MkDocs. 

In defense of `index_page` lots of web servers let you configure the default page. Both apache and nginx provide this flexibility, why not MkDocs? 

I'm happy to throw a PR your way, but if this an ideological / conventions battle that you guys aren't open to, I don't want to waste anyone's time. Thanks for the great tool! I'd really appreciate your consideration on this issue, it would help me a lot.

**Reply by squidfunk** [reply]
Author of [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/) here. I'm not sure if what you proposed is a use case the maintainers of MkDocs would consider, but for us it is definitely one, as we explained in our [latest blog post](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/), specifically the case of [information silos](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/?h=information+silos#editing-and-collaboration). If you'd be up for a chat, we'd love to learn about how your company uses MkDocs (even if it doesn't use Material for MkDocs).

Feel free to drop me a message at martin.donath@squidfunk.com, if you want to talk about this ☺️

**Reply by pawamoy** [reply]
@klvs since you're building docs in CI/CD with your own scripting, couldn't you rename readme files into index ones, or symlink them somehow? Also, do I understand correctly that you declare a `nav` for each project? I believe MkDocs uses README.md by default as index page when you don't specify the navigation. Other plugins like awesome-pages might help you do both, have you looked into existing plugins :) ?

**Reply by lovelydinosaur** [reply]
Thanks for the insight into how your teams are working with MkDocs, Alex. I'm also interested in connecting, to understand more about how we can be building out the project.

Either separately or perhaps in a group call if that'd suit you both? Welcome to continue this over at tom@tomchristie.com

**Reply by klvs** [reply]
@squidfunk I realize my use case is a little different from what mkdocs is meant for but it _does_ work really well for it. I'll take a look at Material for MkDocs.

@pawamoy You are right, it does. I now realize what I was trying to do is get the root level readme set as the index rather than `docs/readme.md`. My bad. This is actually more of https://github.com/mkdocs/mkdocs/discussions/3062 than it is this discussion now that I think about it.

Your suggestion of renaming or symlinking might be what I end up doing. It's just that if I move forward with mkdocs I might be setting up a LOT of mkdocs instances and I'm trying to minimize custom config or at least keep it contained to a config file. That's more my problem than MkDocs' but being able to specify and index_page would be nice...

@tomchristie Sure! I'll shoot you a message.

**Reply by squidfunk** [reply]
That's fair! I'll let you two fiddle out the specifics ☺️

**Comment by noelmcloughlin** [comment]
Just looping back to my original issue, I agree "index" is implementation
detail in Web 1.0 WWW servers, but It has real world meaning for Web 2.0,
i.e. my use case was linked data app (https://www.w3.org/wiki/LinkedData),
where 'index' was named of thing and Tim burners lee said URIs should be
meaningful. I have moved on, so not bothered anymore, but advocating for
flexibility.
